packages/cli/src/cmds/agentInstaller/agentProcedure.ts

```diff
@@ -108,6 +107,222 @@ export default abstract class AgentProcedure {
+      try {
+        const stdout = runSync(new CommandStruct('git', ['diff', file], this.installer.path));
+        if (stdout.length > 0) {
+          gitDiff = stdout.split('\n')[0];
+          thereWasADiff = true;
+        } else {
+          gitDiff = '[no diff]';
+        }
+      } catch (e) {
+        const gitError = (e as Error).message.split('\n')[1];
+        gitDiff = `[git diff failed, ${gitError}]`;
+      }
+
+      if (thereWasADiff) {
+        filesThatMatched.push(file);
+      }
+    }
+
+    return filesThatMatched;
+  }
+
+  async filesArePartOfProject(files: string[]): Promise<string[]> {
+    let filesThatMatched: string[] = [];
+
+    for (const file of files) {
+      let gitStatus;
+      let isPartOfProject: boolean = false;
+      try {
+        const stdout = runSync(
+          new CommandStruct('git', ['status', '-s', file], this.installer.path)
+        );
+        if (stdout.length > 0) {
+          gitStatus = stdout.split('\n')[0];
+          if (gitStatus[0] !== '?') isPartOfProject = true;
+        } else {
+          gitStatus = '[no status]';
+        }
+      } catch (e) {
+        const gitError = (e as Error).message.split('\n')[1];
+        gitStatus = `[git status failed, ${gitError}]`;
+      }
+
+      if (isPartOfProject) {
+        filesThatMatched.push(file);
+      }
+    }
+
+    return filesThatMatched;
+  }
+
+  async gitAdd(files: string[]): Promise<boolean> {
+    let gitAdd;
+    let isPartOfProject: boolean = false;
+    let itWorked: boolean = false;
+    try {
+      let params: string[] = ['add'];
+      for (const file of files) {
+        params.push(file);
+      }
+      const stdout = runSync(new CommandStruct('git', params, this.installer.path));
+      itWorked = true;
+    } catch (e) {
+      const gitError = (e as Error).message.split('\n')[1];
+      gitAdd = `[git add failed, ${gitError}]`;
+    }
+
+    return itWorked;
+  }
+
+  async gitCommit(files: string[], commitMessage: string): Promise<boolean> {
+    let gitCommit;
+    let isPartOfProject: boolean = false;
+    let itWorked: boolean = false;
+    try {
+      let params: string[] = ['commit', '-m', commitMessage];
+      for (const file of files) {
+        params.push(file);
+      }
+      const stdout = runSync(new CommandStruct('git', params, this.installer.path));
+      itWorked = true;
+    } catch (e) {
+      const gitError = (e as Error).message.split('\n')[1];
+      gitCommit = `[git commit failed, ${gitError}]`;
+    }
+
+    return itWorked;
+  }
+
+  async configurationFiles() {
+    let filesWithInstaller: fileWithInstaller[] = [
+      {
+        name: 'AppMap',
+        file: 'appmap.yml',
+      },
+      { name: this.installer.name, file: this.installer.buildFile },
+    ];
+    // this is specific to Ruby
+    if (this.installer.name === 'Bundler')
+      filesWithInstaller.push({
+        name: this.installer.name,
+        file: 'Gemfile.lock',
+      });
```

Comment:
Although the `buildFile` for Ruby is only `Gemfile`, we should also commit `Gemfile.lock` because this file changes too and people usually commit it too.

I thought of adding an extra get-ter in each installer called something like `configurationFiles` that returns an array of all files for each language.  WDYT?

Reply:
Changed it to look at git status before and after we install.

Also verified it installs in sub-directories.